tests/TestCase/Database/QueryTest.php

```diff
@@ -4240,12 +4240,8 @@ public function testSqlCaseStatement()
 
         // Postgres requires the case statement to be cast to a integer
         if ($this->connection->getDriver() instanceof \Cake\Database\Driver\Postgres) {
-            $publishedCase = $query->func()
-                ->cast([$publishedCase, 'integer' => 'literal'])
```

Comment:
Does this form now fail? In order to maintain backwards compatibility we should also continue to accept this call signature.

Reply:
I added backward compatibility with deprecation warning message.